File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        vals = {}
        for num in nums:
            vals[num] = vals.get(num, 0)+1
        buckets = [[] for _ in range(len(nums) + 1)]
        for key, value in vals.items():
            buckets[value].append(key)
        ans = []
        for freq in range(len(buckets)-1, 0, -1):
            for num in buckets[freq]:
                ans.append(num)
                if len(ans) == k:
                    return ans

        return ans

File: Data Structures & Algorithms/two-integer-sum/submission-1.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> vals = new HashMap<>();
        int[] ans = new int [2];
        for (int i = 0; i<nums.length; i++){
            if(vals.containsKey(target-nums[i])){
                ans[0] = vals.get(target-nums[i]);
                ans[1] = i;
            }
            else{
                vals.put(nums[i], i);
            }
        }
        return ans;
    }
}
